docs/reference/core-concepts.md

```diff
@@ -7,3 +7,73 @@ title: Great Expectations Core Concepts
 # Data Docs
 
 # Profiling
+
+Great Expectations provides a mechanism to automatically generate expectations, using a feature called a **Profiler**. A Profiler builds an Expectation Suite from one or more Data Assets. It may also validates the data against the newly-generated Expectation Suite to return a Validation Result. There are several Profilers included with Great Expectations.
+
+A Profiler makes it possible to quickly create a starting point for generating expectations about a Dataset. For example, during the init flow, Great Expectations currently uses the UserConfigurableProfiler to demonstrate important features of Expectations by creating and validating an Expectation Suite that has several different kinds of expectations built from a small sample of data. A Profiler is also critical to generating the Expectation Suites used during profiling.
+
+## Rule-based Profilers
+
+**Rule-based profilers** allow users to provide a highly configurable specification which is composed of **Rules** to use in order to build an **Expectation Suite** by profiling existing data.
+
+Imagine you have a table of Sales that comes in every month. You could profile last month's data, inspecting it in order to automatically create a number of expectations that you can use to validate next month's data.  
+
+A **Rule** in a rule-based profiler could say something like "Look at every column in my Sales table, and if that column is numeric, add an `expect_column_values_to_be_between` expectation to my Expectation Suite, where the `min_value` for the expectation is the minimum value for the column, and the `max_value` for the expectation is the maximum value for the column."
+
+Each rule in a rule-based profiler has three types of components:
+
+1. **DomainBuilders**: A DomainBuilder will inspect some data that you provide to the Profiler, and compile a list of Domains for which you would like to build expectations. 
+1. **ParameterBuilders**: A ParameterBuilder will inspect some data that you provide to the Profiler, and compile a dictionary of Parameters that you can use when constructing your ExpectationConfigurations
+1. **ExpectationConfigurationBuilders**: An ExpectationConfigurationBuilder will take the Domains compiled by the DomainBuilder, and assemble ExpectationConfigurations using Parameters built by the ParameterBuilder
+
+In the above example, imagine your table of Sales has twenty columns, of which five are numeric:
```

Comment:
What are you thoughts on including a snippet of this example table here? Is there value in clarifying the data source or is that unnecessary? I'm leaning towards the latter but figured it would be worth bringing up.

Reply:
Nope I'm in agreement with you after a bit of thought! For a user to get this far in their GE journey, they probably know a thing or two about realistic datasets :)